.github/workflows/push-to-gar-docker-multiarch.yaml

```diff
@@ -0,0 +1,131 @@
+name: Multi-platform Docker image push to Google Artifact Registry
+
+on:
+  workflow_call:
+    inputs:
+      platforms:
+        default: "['amd64']"
+        type: string
+      image_name:
+        required: true
+        type: string
+      registry:
+        description: |
+          Google Artifact Registry to store docker images in.
+        default: "us-docker.pkg.dev"
+        type: string
+      tags:
+        type: string
+        description: |
+          List of Docker images to be pushed.
+        required: true
+      context:
+        description: |
+          Path to the Docker build context.
+        default: "."
+        type: string
+      environment:
+        description: |
+          Environment for pushing artifacts (can be either dev or prod).
+        default: dev
+        type: string
+      build-args:
+        type: string
+        description: |
+          List of arguments necessary for the Docker image to be built.
+        default: ""
+      file:
+        type: string
+        description: |
+          The dockerfile to use.
+        required: false
+      cache-from:
+        type: string
+        description: |
+          Where cache should be fetched from
+        required: false
+        default: "type=gha"
+      cache-to:
+        type: string
+        description: |
+          Where cache should be stored to
+        required: false
+        default: "type=gha,mode=max"
+      ssh:
+        type: string
+        description: |
+          List of SSH agent socket or keys to expose to the build
+
+permissions:
+  contents: read
+  id-token: write
+
+env:
+  platform_env: ${{ inputs.platforms }}
+
+jobs:
+  build-multi-arch:
+    outputs:
+      full_image_name: ${{ steps.build-multiarch.outputs.full_image_name }}
+    strategy:
+      matrix:
+        os: ${{ fromJson(inputs.platforms ) }}
+        tag: ${{ fromJson(inputs.tags ) }}
+    runs-on: ubuntu-${{ matrix.os }}-large
+    name: build-${{ matrix.os }}-image-for-tag-${{ matrix.tag }}
+    steps:
+      - name: Checkout
+        env:
+          action_repo: ${{ github.action_repository }}
+          action_ref: ${{ github.action_ref }}
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          repository: ${{ env.action_repo }}
+          ref: ${{ env.action_ref }}
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          repository: grafana/shared-workflows
+          ref: multiarch-docker-builds-using-sh-runners
+          path: shared-workflows
```

Comment:
Aren't these the same? I thought `github.action_repository` will be `shared-workflows` when you're running this, ditto `action_ref` and the selected ref? Or is it different for reusable workflows?